### DISCUSSION TITLE
[Suggestion] Make ActiveJ as microservice solution instead just another SpringBoot like framework

### DISCUSSION THREAD

**Discussion by gsitgithub** [activej/activej#227] [Q&A] [answered]
Hope you are doing well. This is a suggestion.
Most of java web application use Spring/SpringBoot framework even it is slower than ActiveJ, because it provides solution to common problems. Currently common problem is microservices.
Could you have a look at golang repo https://github.com/zeromicro/go-zero as microservice solution. 
Can ActiveJ be like this for Java as a microservice solution not just another framework or library?
An example repository of activej microservices with load balancing, builtin concurrency control, adaptive circuit breaker, adaptive load shedding, auto recovery, call tracing, metrics
 could add more help to examples in activej. 
Thanks.

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @gsitgithub

ActiveJ is not just a framework, but a general-purpose platform consisting of various components, including Dependency Injection and Asynchronous I/O.
 
Even our HTTP module is implemented as a self-sufficient solution, and, unlike other frameworks, is not based on third-party I/O libraries.
 
As for the microservices technology you're interested in, we have something similar in the form of ActiveJ RPC. You can develop different microservices strategies on top of ActiveJ RPC.
 
The main goal of ActiveJ is to maximize performance while ensuring consistency and compatibility of each individual component, (for example, multithreading model and asynchronous model) .
